File: src/main/java/com/example/game/units/Archer.java
package com.example.game.units;

import com.example.game.core.Point2D;
import com.example.game.core.Unit;

public class Archer extends Unit {
    public Archer(int id, Point2D startPos) {
        super("궁수", id, false, startPos);
    }

    @Override
    public void moveTo(Point2D dest) {
        log("걸어서 " + position + " -> " + dest + " 로 이동합니다. (느리게)");
        this.position = dest;
    }

    @Override
    protected boolean canAttack(Unit target) {
        // 궁수는 "공중/지상 모두 공격 가능"
        return true;
    }
}

File: src/main/kotlin/com/example/game/units/Archer.kt
package com.example.game.units

import com.example.game.core.Point2D
import com.example.game.core.UnitBase
import com.example.game.traits.Attackable
import com.example.game.traits.Movable

/**
 * Archer(궁수): 걸어서 이동하며 화살로 지상/공중 모두 공격 가능.
 */
class Archer(id: Int, start: Point2D) :
    UnitBase(id = id, displayName = "궁수", position = start, isFlying = false),
    Movable, Attackable {

    override fun moveTo(dest: Point2D) {
        log("걸어서 $position → $dest 로 이동합니다. (기사는 말, 궁수는 도보로 더 느림)")
        position = dest
    }

    override fun attack(target: UnitBase) {
        log("화살을 쏘아 ${target.displayName}#${target.id} 를 공격합니다! (지상/공중 모두 가능)")
    }
}